docs/eom_base.rst

```diff
@@ -0,0 +1,255 @@
+    >>> from numpy import linspace
+    >>> from pydy.system import System
+    >>> from sympy import symbols, Matrix
+    >>> from sympy.physics.mechanics import dynamicsymbols, eombase
+
+The first step is to define the dynamic and constant symbols that will be used
+in the system. ::
+
+    >>> x1, x2, u1, u2 = dynamicsymbols('x1 x2 u1 u2')
+    >>> m1, c1, k1 = symbols('m1 c1 k1')
+    >>> m2, c2, k2 = symbols('m2 c2 k2')
+
+Next step is to define the equations of motion in multiple forms:
+
+[1] x' = F(x, t, r, p)
+
+[2] M(x, p) x' = F(x, t, r, p)
+
+[3] M(q, p) u' = F(q, u, t, r, p)
+    q' = G(q, u, t, r, p) ::
+
+    >>> mm = Matrix([[m1,  0],
+    ...             [0,  m2]])
+    >>> f = Matrix([-(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+    ...             k2*x1 - k2*x2 + c2*u1 - c2*u2])
+    >>> mm_full = Matrix([[1, 0,  0,  0],
+    ...                   [0, 1,  0,  0],
+    ...                   [0, 0, m1,  0],
+    ...                   [0, 0,  0, m2]])
+    >>> f_full = Matrix([u1,
+    ...                  u2,
+    ...                  -(k1+k2)*x1 + k2*x2 - (c1+c2)*u1 + c2*u2,
+    ...                  k2*x1 - k2*x2 + c2*u1 - c2*u2])
+    >>> G = Matrix([u1, u2])
+    >>> RHS = mm_full.LUsolve(f_full)
+
+Now the various forms of varible entry to `eombase` will be initiated ::
+
+    >>> coordinates = (x1, x2)
+    >>> speeds = (u1, u2)
+    >>> states = (x1, x2, u1, u2)
+
+This example will also show `system`'s ability to calculate an arbitrary user
+defined functions along with the states. Specifically the kinetic energy of the
+whole system is going to be determined. ::
+
+    >>> KE = 0.5 * (m1*u1**2 + m2*u2**2)
+    >>> out_eqns = {"kinetic energy": KE}
+
+The `eombase` instances are now ready to be initialized. To showcase possible
+input combinations, all three of the equations of motion forms will be used. ::
+
+    >>> eom1 = eombase.EOM(coordinates, RHS, speeds=speeds, 
+    ...                    output_eqns=out_eqns)
+    >>> eom2 = eombase.EOM(states, f_full, mass_matrix_full=mm_full,
+    ...                     num_coordinates=2)
+    >>> eom3 = eombase.EOM(states, f, mass_matrix=mm, coordinate_derivatives=G)
+
+The system instance will now be initialized and set up to perform simulation. ::
+
+    >>> sys = System(eom1)
+    >>> sys.times = linspace(0, 10, num=100)
+    >>> sys.constants = {m1: 10.0, c1: 10.0, k1: 10.0, m2: 5.0, c2: 5.0, k2:
+    ...                  5.0}
+    >>> sys.initial_conditions = {x1: 1.0, x2: 0.0, u1: 0.0, u2: 0.0}
+
+Now the actual system simulation is ready to be run. ::
+
+    >>> out = sys.integrate()
+    >>> out
+    array([[1.0, 0.0, 0.0, 0.0], [...], ...])
+
+The user specified output equations should be able to be determine now that the
+time simulation of the states is complete. ::
+
+    >>> sys.output_eqns()
+    {"kinetic energy": array([[0.0], [...], ...])}
+    >>> eom1.output_eqns_results
+    {"kinetic energy": array([[0.0], [...], ...])}
```

Comment:
Sounds good, These will eventually  be translated to unit tests that will ensure it can work with a variety of inputs.